cli/command.go

```diff
@@ -0,0 +1,295 @@
+package cli
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/cheynewallace/tabby"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/jasper"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+// RunCMD provides a simple user-centered command-line interface for
+// running commands on a remote instance.
+func RunCMD() cli.Command {
+	const (
+		commandFlagName = "command"
+		envFlagName     = "env"
+		sudoFlagName    = "sudo"
+		sudoAsFlagName  = "sudoAs"
+		idFlagName      = "id"
+		noShellFlagName = "noShell"
+		tagFlagName     = "tag"
+		waitFlagName    = "wait"
+	)
+
+	return cli.Command{
+		Name:  "run",
+		Usage: "Run a command with Jasper",
+		Flags: append(clientFlags(),
+			cli.StringSliceFlag{
+				Name:  joinFlagNames(commandFlagName, "c"),
+				Usage: "specify a command to run on a remote jasper service. may specify more than once",
+			},
+			cli.StringSliceFlag{
+				Name:  envFlagName,
+				Usage: "specify environment variables, in '<key>=<val>' forms. may specify more than once",
+			},
+			cli.BoolFlag{
+				Name:  sudoFlagName,
+				Usage: "use this flag to run the command with sudo",
+			},
+			cli.StringFlag{
+				Name:  sudoAsFlagName,
+				Usage: "use this to run commands as another user as in 'sudo -u <user>'",
+			},
+			cli.StringFlag{
+				Name:  idFlagName,
+				Usage: "specify an id for this process (optional)",
+			},
+			cli.BoolTFlag{
+				Name:  noShellFlagName,
+				Usage: "when specified execute commands directly without shell. If multiple commands are specified they do not share a shell environment",
+			},
+			cli.StringSliceFlag{
+				Name:  tagFlagName,
+				Usage: "specify one or more tag names for the process",
+			},
+			cli.BoolFlag{
+				Name:  waitFlagName,
+				Usage: "specify to block until the process returns (subject to service timeouts), propogating the exit code from process",
```

Comment:
The command will block until `Run()` completes by defautl unless you specify `(*Command).Background(true)`. `Wait()` will only be guaranteed to work if it's not running in the background (i.e. what it's doing now).